modules/oe_translation_local/modules/oe_translation_multivalue/src/MultivalueTranslationIdGenerator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_multivalue;
+
+use Drupal\Component\Uuid\UuidInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Generates a new ID for a given field value translation_id.
+ */
+class MultivalueTranslationIdGenerator {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The UUID service.
+   *
+   * @var \Drupal\Component\Uuid\UuidInterface
+   */
+  protected $uuid;
+
+  /**
+   * Constructs a new MultivalueTranslationIdGenerator.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Component\Uuid\UuidInterface $uuid
+   *   The UUID service.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, UuidInterface $uuid) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->uuid = $uuid;
+  }
+
+  /**
+   * Generates a UUID for the translation value.
+   *
+   * We also do a check to make sure the UUID doesn't exist already in the DB.
+   *
+   * @param string $field_id
+   *   The field ID in the format entity_type.field_name.
+   *
+   * @return string
+   *   The UUID.
+   */
+  public function generateTranslationUuid(string $field_id): string {
+    [$entity_type_id, $field_name] = explode('.', $field_id);
+    $table_mapping = $this->entityTypeManager->getStorage($entity_type_id)->getTableMapping();
+    $table = $table_mapping->getFieldTableName($field_name);
+    $column = $field_name . '_translation_id';
+    $exists = TRUE;
+    while ($exists) {
+      $uuid = $this->uuid->generate();
+      $query = sprintf("SELECT %s FROM {%s} WHERE %s = '%s'", $column, $table, $column, $uuid);
+      $exists = !empty(\Drupal::database()->query($query)->fetchAll());
```

Comment:
We may inject `database` service.
also, we may use `$this->database->select()`.